test/famfs_unit.cpp

```diff
@@ -453,4 +457,88 @@ TEST(famfs, famfs_log)
 
 	rc = famfs_fsck_scan(sb, logp, 1, 3);
 	ASSERT_EQ(rc, 0);
+
+	rc = famfs_fsck("/tmp/famfs/.meta/.superblock", 1 /* mmap */, 1, 1);
+	ASSERT_EQ(rc, 0);
+	rc = famfs_fsck("/tmp/famfs/.meta/.superblock", 0 /* read */, 1, 1);
+	ASSERT_EQ(rc, 0);
+	rc = famfs_fsck("/tmp/nonexistent-file", 0 /* read */, 1, 1);
+	ASSERT_NE(rc, 0);
+
+	/* Save good copies of the log and superblock */
+	system("cp /tmp/famfs/.meta/.log /tmp/famfs/.meta/.log.save");
+	system("cp /tmp/famfs/.meta/.superblock /tmp/famfs/.meta/.superblock.save");
+
+	truncate("/tmp/famfs/.meta/.superblock", 8192);
+	rc = famfs_fsck("/tmp/famfs/.meta/.superblock", 0 /* read */, 1, 1);
+	ASSERT_EQ(rc, 0);
+
+	truncate("/tmp/famfs/.meta/.superblock", 7);
+	rc = famfs_fsck("/tmp/famfs/.meta/.superblock", 0 /* read */, 1, 1);
+	ASSERT_NE(rc, 0);
+
+	truncate("/tmp/famfs/.meta/.log", 8192);
+	rc = famfs_fsck("/tmp/famfs/.meta/.superblock", 0 /* read */, 1, 1);
+	ASSERT_NE(rc, 0);
+
+	unlink("/tmp/famfs/.meta/.log");
+	rc = famfs_fsck("/tmp/famfs/.meta/.superblock", 0 /* read */, 1, 1);
+	ASSERT_NE(rc, 0);
+	rc = famfs_fsck("/tmp/famfs/.meta/.superblock", 1 /* mmap */, 1, 1);
+	ASSERT_NE(rc, 0);
+	unlink("/tmp/famfs/.meta/.superblock");
+	rc = famfs_fsck("/tmp/famfs/.meta/.superblock", 0 /* read */, 1, 1);
+	ASSERT_NE(rc, 0);
+	rc = famfs_fsck("/tmp/famfs/.meta/.superblock", 1 /* mmap */, 1, 1);
+	ASSERT_NE(rc, 0);
+
+	system("chmod 200 /tmp/famfs/.meta/.log");
+	rc = famfs_fsck("/tmp/famfs/.meta/.log", 1 /* mmap */, 1, 1);
+	ASSERT_NE(rc, 0);
+	rc = famfs_fsck("/tmp/famfs/.meta/.log", 0 /* read */, 1, 1);
+	ASSERT_NE(rc, 0);
+
+	system("chmod 200 /tmp/famfs/.meta/.superblock");
+	rc = famfs_fsck("/tmp/famfs/.meta/.superblock", 1 /* mmap */, 1, 1);
+	ASSERT_NE(rc, 0);
+	rc = famfs_fsck("/tmp/famfs/.meta/.superblock", 0 /* read */, 1, 1);
+	ASSERT_NE(rc, 0);
+
+	system("cp /tmp/famfs/.meta/.log.save /tmp/famfs/.meta/.log");
+	system("cp /tmp/famfs/.meta/.superblock.save /tmp/famfs/.meta/.superblock");
+
+	rc = famfs_release_locked_log(&ll);
+	ASSERT_EQ(rc, 0);
+
+	system("chmod 444 /tmp/famfs/.meta/.log"); /* log file not writable */
+
+	mock_role = FAMFS_CLIENT;
+	rc = famfs_init_locked_log(&ll, "/tmp/famfs", 1);
+	ASSERT_NE(rc, 0);
+
+	mock_role = FAMFS_CLIENT;
+	rc = famfs_init_locked_log(&ll, "/tmp/famfs", 1);
+	ASSERT_NE(rc, 0);
+
+	mock_role = 0;
+
+#if 0
```

Comment:
Probably no need to keep it, if it's not working, we could create an issue to cover this ?